### PR TITLE
use the lmdb-master.3 branch

### PR DESCRIPTION
**Pull request by irevoire**
After investigating https://github.com/meilisearch/meilisearch/issues/3017, we found out that it was due to lmdb and that, without any code change on our side, bumping using the lmdb-master-3 branch fix our issues.

But, we’re not really confident about what changed between the `mdb.master` and `mdb.master3` branches; thus this is a temporary change, and we hope we’ll be able to move to the new version of heed asap (either before the end of the pre-release or for the next release).

--------

The bug is hard to reproduce; I can reproduce it 100% of the time on my archlinux personal computer. But on a scaleway archlinux bare-metal machine, it doesn’t reproduce. It’s flaky on our test suite, but @loiclec was able to write a minimal test that reproduces it every time on macOS.
Basically, what happens is when there are multiple threads opening databases in a different directory at the same time.
If there are 10 or more threads running at the same time, lmdb starts throwing the `Invalid argument (os error 22)` error for no reason, we believe.
I would like to submit an issue to lmdb, but I don’t really have the time to write a test in C without heed currently.

@hyc, if you want to take a look at it, here is the repo that reproduces the issue on macOS: https://github.com/irevoire/heed-bug

### CODE REVIEW

**Review by Kerollmops — APPROVED** [review]
Ok, so, we will merge this PR and use it in Meilisearch for now, even if we would have liked to do that in a future version of the engine. But, there is an advantage to doing so the earliest possible: we can test `lmdb-master-3` before shipping it in v1.0 of the engine 🎉
bors merge
